data-models/api/v1alpha1/healthstatus_types.go

```diff
@@ -0,0 +1,118 @@
+
+package v1alpha1
+
+import (
+	"github.com/nvidia/nvsentinel/data-models/pkg/model"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +kubebuilder:validation:Enum=NotStarted;InProgress;Failed;Succeeded;AlreadyDrained;UnQuarantined;Quarantined;AlreadyQuarantined;Cancelled
+type Status = model.Status
+
+type OperationStatus = model.OperationStatus
+
+// HealthEventSnapshot represents a read-only snapshot of a reported health event.
+// This data is observational and originates outside Kubernetes.
+type HealthEventSnapshot struct {
+	// Version of the reported health event schema
+	Version uint32 `json:"version,omitempty"`
+
+	// Reporting agent that generated the event
+	Agent string `json:"agent,omitempty"`
+
+	// Component class that raised the event
+	ComponentClass string `json:"componentClass,omitempty"`
+
+	// Specific check or rule that triggered the event
+	CheckName string `json:"checkName,omitempty"`
+
+	// Indicates whether the event is fatal
+	IsFatal bool `json:"isFatal,omitempty"`
+
+	// Indicates whether the system was reported healthy
+	IsHealthy bool `json:"isHealthy,omitempty"`
+
+	// Human-readable event message
+	Message string `json:"message,omitempty"`
+
+	// Recommended action provided by the reporting system
+	RecommendedAction string `json:"recommendedAction,omitempty"`
+
+	// Error codes associated with this health event
+	ErrorCode []string `json:"errorCode,omitempty"`
+
+	// Additional key-value metadata provided by the agent
+	Metadata map[string]string `json:"metadata,omitempty"`
+
+	// Time at which the event was generated by the source
+	GeneratedTimestamp *metav1.Time `json:"generatedTimestamp,omitempty"`
+}
+
+// RemediationStatus captures the observed state of remediation workflows
+type RemediationStatus struct {
+	// Indicates whether the node is quarantined
+	NodeQuarantined *Status `json:"nodeQuarantined,omitempty"`
+
+	// Status of user pods eviction process
+	UserPodsEvictionStatus *OperationStatus `json:"userPodsEvictionStatus,omitempty"`
+
+	// Whether the fault has been remediated
+	FaultRemediated *bool `json:"faultRemediated,omitempty"`
+
+	// Timestamp of the last remediation attempt
+	LastRemediationTimestamp *metav1.Time `json:"lastRemediationTimestamp,omitempty"`
+}
+
+// HealthEventStatus defines the observed state of HealthStatus
+type HealthEventStatus struct {
+	// Snapshot of the reported health event
+	Event *HealthEventSnapshot `json:"event,omitempty"`
+
+	// Observed remediation state
+	Remediation *RemediationStatus `json:"remediation,omitempty"`
+}
+
+// HealthEventSpec defines the desired state of HealthStatus
+type HealthEventSpec struct {
+	// Unique identifier for the health event
+	// +kubebuilder:validation:Required
+	EventID string `json:"eventID"`
+
+	// Node associated with this health event
+	// +kubebuilder:validation:Required
+	NodeName string `json:"nodeName"`
+}
+
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+type HealthStatus struct {
```

Comment:
so my thought was this struct would look like this:

```
// +kubebuilder:object:root=true
// +kubebuilder:subresource:status
type HealthStatus struct {
  metav1.TypeMeta   `json:",inline"`
  metav1.ObjectMeta `json:"metadata,omitempty"`

  Spec   model.HealthEvent   `json:"spec,omitempty"`
  Status model.HealthEventStatus `json:"status,omitempty"`
}
```

So you would not define your own spec and status objects we would just use the existing ones.

There's some implications for api-versioning going forward if we want to adhere to best practices that we should discuss with the NVIDIA folks if we do it this way but it seems like a clean way to share the object that the other datasources use.